controllers/ProposalController.php

```diff
@@ -27,6 +55,91 @@ public function actionMyProposals(): string
         ]);
     }
 
+    /**
+     * Check if a Proposal exists. If true it returns the proposal.
+     * If false it throws an exception.
+     *
+     * @param int $id
+     * @return Proposal|null
+     * @throws \Exception
+     */
+    private function checkIfProposalExists(int $id): ?Proposal
+    {
+        $unauthorizedException = NotFoundHttpException::class;
+
+        if (!is_null($selectedProposal = Proposal::findOne(['id' => $id]))) {
+            return $selectedProposal;
+        }
+        throw new $unauthorizedException();
+    }
+
+    /**
+     * Check if User is owner of the proposal.
+     * If true it returns nothing letting the process continue.
+     * If false it throws an exception
+     *
+     * @param int $submitterId
+     * @throws Exception
+     */
+    private function checkIfUserIsOwnerOfProposal(int $submitterId)
+    {
+        if($submitterId == self::getCurrentUser()->id) {
+            return;
+        }
+        throw new Exception('Not owner of this proposal');
```

Comment:
see line 13 : the Exception you imported is the yii\db\Exception. You should throw an UnauthorizedException, or simply return false as it's a private function

Reply:
It's a temporary Exception waiting for creating a customized exception for that case. I make the change in case we haven't time to design it.